src/Tutor.Core/Domain/KnowledgeMastery/KnowledgeComponentMastery.cs

```diff
@@ -176,7 +177,11 @@ public Result RecordInstructorMessage(string message)
     protected override void Apply(DomainEvent @event)
     {
         if (@event is not KnowledgeComponentEvent kcEvent) throw new EventSourcingException("Unexpected event type: " + @event.GetType());
-
+        if (kcEvent is SessionPaused)
```

Comment:
Don't add a special case here. Modify the timestamp at the place where the event is created.